src/os/bluestore/BlueStore.cc

```diff
@@ -6546,7 +6551,15 @@ int BlueStore::open_db_environment(KeyValueDB **pdb, bool to_repair)
 
 int BlueStore::close_db_environment()
 {
-  _close_db_and_around();
+  // verify that the db is actually closed
+  if (db) {
+    RocksDBStore* rocks_db = dynamic_cast<RocksDBStore*>(db);
```

Comment:
if you really want to call is_open() in production - which is questionable as it's called exclusively  from ceph_assert - it worth to make is_open() virtual and do not use dynamic_cast

Reply:
How about skipping the check altogether?

Reply:
sounds good